ev3dev/core.py

```diff
@@ -182,8 +183,16 @@ def get_index(file):
                 self._device_index = get_index(name)
                 self.connected = True
             except StopIteration:
+                self._path = None
+                self._device_index = None
                 self.connected = False
 
+    def __str__(self):
```

Comment:
Do you mean why is it here instead of in helper.py or what does __str__() do?  Sorry, I'm not following what the question/concern is.